src/server/info_collector.cpp

```diff
@@ -143,7 +143,7 @@ void info_collector::on_app_stat()
 {
     ddebug("start to stat apps");
     std::map<std::string, std::vector<row_data>> all_rows;
-    if (!get_app_partition_stat(&_shell_context, all_rows)) {
+    if (!get_app_partition_stat(_shell_context.get(), all_rows)) {
```

Comment:
you can change the type for first param of  get_app_partition_stat to std::shared_ptr. avoid to use raw pointer

Reply:
+1, and I recommend that raise another pull request to do it, including update `_shell_context` to std::shared_ptr, and update functions like `get_app_partition_stat`

Reply:
I will solve it in the other PR

Reply:
There is no necessary to replace the raw pointer to shared_ptr, because `_shell_context` will not be destructed under any circumstances.